src/app/pages/onboarding/set-password/components/password-field.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { FiEye, FiEyeOff } from 'react-icons/fi';
+import { useField } from 'formik';
+import { Box, Button, color, Flex, Input } from '@stacks/ui';
+
+import { ValidatedPassword } from '@app/common/validation/validate-password';
+import { Caption } from '@app/components/typography';
+import { OnboardingSelectors } from '@tests/integration/onboarding/onboarding.selectors';
+
+import { PasswordStrengthIndicator } from './password-strength-indicator';
+import { getIndicatorsOfPasswordStrength } from './password-field.utils';
+
+interface PasswordFieldProps {
+  strengthResult: ValidatedPassword;
+}
+export function PasswordField({ strengthResult }: PasswordFieldProps) {
+  const [field] = useField('password');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const { strengthColor, strengthText } = getIndicatorsOfPasswordStrength(strengthResult);
+
+  return (
+    <>
+      <Box position="relative">
+        <Input
+          _focus={{ border: '2px solid #5546FF' }}
+          autoFocus
+          border="2px solid"
+          data-testid={OnboardingSelectors.NewPasswordInput}
+          height="64px"
+          key="password-input"
+          placeholder="Set a password"
+          type={showPassword ? 'text' : 'password'}
+          {...field}
+        />
+        <Button
+          _focus={{ bg: 'transparent', boxShadow: 'none' }}
+          _hover={{ bg: 'transparent', boxShadow: 'none' }}
+          bg="transparent"
+          boxShadow="none"
+          color={color('text-title')}
+          height="20px"
+          onClick={
+            field.value
+              ? evt => {
+                  evt.preventDefault();
+                  setShowPassword(!showPassword);
+                }
+              : undefined
+          }
```

Comment:
Rather than conditionally toggling the presence of a fn, it'd make more sense to me to do this check at runtime than render time

```suggestion
          onClick={evt => {
            evt.preventDefault();
            if (field.value) setShowPassword(!showPassword);
          }}
```

Reply:
Further to submit issue, I don't see any reason why a user isn't within their right to first, with an empty input, decide to show their password? 

Why would we go out of our way to prevent this, just because the field is empty?

Reply:
I guess the button just toggles icons but nothing changes in the input field bc nothing is there, but I can remove it.

Reply:
> I guess the button just toggles icons but nothing changes in the input field bc nothing is there, but I can remove it. 

Both scenarios are legit:
- User types password hidden, then chooses to see it with toggle
- User toggles to see password, then types it

The latter is pretty unlikely, I'd expect, but that isn't a reason to prevent it from being possible